internal/controller/controllers/agent_controller.go

```diff
@@ -572,6 +612,16 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 						ret = ctrl.Result{RequeueAfter: r.ApproveCsrsRequeueDuration}
 					}
 				}
+				if err = r.applyDay2NodeLabels(ctx, log, agent, spokeClient); err != nil {
+					log.WithError(err).Errorf("Failed to apply labels for day2 node %s/%s", agent.Namespace, agent.Name)
+					return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, nil
+				}
+				if h.Progress.CurrentStage != agent.Status.Progress.CurrentStage {
```

Comment:
Not sure i understand which status we should set here? Or you just moved it from another place?

Reply:
Changed for a simpler flow